### ISSUE TITLE
🐛 [Bug Report] Warlocks can Enslave demons belonging to other NPCs

### ISSUE THREAD

**Issue by Nytanath** [cmangos/issues#2766]
### Bug Details

Warlocks should not be able to cast Enslave Demon on a demon that's summoned by a NPC.

Proof:

https://user-images.githubusercontent.com/99603810/155040419-9883ddba-a7e3-4b0d-985f-7c76f038134e.mp4

### Steps to Reproduce

1. Create a Warlock
2. Learn Enslave Demon (.learn 1098)
3. Grab a Soul Shard (.add 6265)
4. .go creature id 3198
5. Try to enslave that NPC's pet.

### Core SHA1 Commit Hash

cbaf1b9cdacbcf61cbb1fa84e3a3620695127c8d

### Database SHA1 Commit Hash

127231a5413902c196e6a976a7a6082c8554674d

### Operating System

Windows 10, x64

### Client Version

1.1.12 (Classic)

**Comment by AnonXS** [comment]
lol nice one.

**Comment by killerwife** [comment]
Test if you can I got it right.

**Comment by Nytanath** [comment]
I don't have a vanilla client on hand right now anymore, but I cherrypicked this on tbc and the bug itself is fixed, thanks! There's only a cosmetic issue now, this check should happen only after the spell finishes casting and not before casting, but it's cosmetic only and way lower priority.
